src/Linker.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+namespace PhpSchool\WorkshopManager;
+
+use Composer\IO\IOInterface;
+use League\Flysystem\Filesystem;
+use PhpSchool\WorkshopManager\Entity\Workshop;
+use PhpSchool\WorkshopManager\Exception\WorkshopNotInstalledException;
+
+/**
+ * Class Linker
+ * @author Michael Woodward <mikeymike.mw@gmail.com>
+ */
+final class Linker
+{
+    /**
+     * @var ManagerState
+     */
+    private $state;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var bool
+     */
+    private $useSystem;
```

Comment:
Not sure why I put it as a property tbh ... but it's just a flag to state whether or not to symlink into the system path as well as the phpschool dir in the users home. 

the idea being that if the phpschool dir is in the users path we don't need to symlink to the system dir.